File: Project/Simulator/Organisms/Plants/SosnowskyHogweed.java
package Simulator.Organisms.Plants;

import Simulator.World;
import Simulator.Organisms.Organism;
import Simulator.Tile;

import java.util.ArrayList;

import org.json.JSONObject;

public class SosnowskyHogweed extends Plant {
    public SosnowskyHogweed(World world) {
        super(10, world, Type.SOSNOWSKY_HOGWEED);
    }

    public SosnowskyHogweed(JSONObject json, World world) {
        super(json, world);
    }

    @Override
    public Plant construct() {
        return new SosnowskyHogweed(world);
    }

    @Override
    public void action() {
        super.action();
        ArrayList<String> killed = new ArrayList<>();
        for (Tile neighbour : tile.getOccupiedNeighbours()) {
            Organism organism = neighbour.getOrganism();
            if (organism instanceof Plant)
                continue;
            // if (organism.getType() == Type.CYBER_SHEEP)
            // continue;

            organism.die();
            if (organism.isDead()) {
                killed.add(organism.getSymbol());
            }
        }
        if (!killed.isEmpty()) {
            StringBuilder log = new StringBuilder(getSymbol() + " killed ");
            for (int i = 0; i < killed.size(); i++) {
                log.append(killed.get(i));
                if (i != killed.size() - 1)
                    log.append(", ");
            }
            world.addLog(log.toString() + "!");
            killed.clear();
        }
    }

    @Override
    public boolean collisionReaction(Organism other) {
        if (other instanceof Plant)
            return false;
        // if (other.getType() == Type.CYBER_SHEEP)
        // return false;
        other.die();
        die();
        world.addLog(other.getSymbol() + " ate " + getSymbol() + " and died!");
        return true;
    }
}

File: Project/Simulator/Organisms/Plants/SosnowskyHogweed.py
import re
from .Plant import Plant
from Simulator.Organisms.Organism import Type, Organism


class SosnowskyHogweed(Plant):
    def __init__(self, world, json=None):
        super().__init__(10, world, Type.SOSNOWSKY_HOGWEED, json)

    def construct(self):
        return SosnowskyHogweed(self._world)

    def action(self):
        super().action()
        killed = []
        if self._tile is None:
            return
        for neighbour in self._tile.getOccupiedNeighbours():
            organism = neighbour.getOrganism()
            if isinstance(organism, Plant):
                continue
            if organism.getType() == Type.CYBER_SHEEP:
                continue

            organism.die()
            if organism.isDead():
                killed.append(organism.getSymbol())
        if not killed:
            return
        log = self.getSymbol() + " killed " + ", ".join(killed)
        self._world.addLog(log + "!")
        killed.clear()

    def collisionReaction(self, other: Organism):
        if isinstance(other, Plant):
            return False
        if other.getType() == Type.CYBER_SHEEP:
            return False
        other.die()
        self.die()
        self._world.addLog(f"{other.getSymbol()} ate " +
                           self.getSymbol() + " and died!")
        return True
